highlevel/robot/gateway/motor.py

```diff
@@ -1,73 +1,156 @@
         """
-        LOGGER.get().debug('gateway_set_speed',
-                           left_speed=left_speed,
-                           right_speed=right_speed)
+        LOGGER.get().debug('motor_gateway_set_target_speeds',
+                           left_speed=tick_left,
+                           right_speed=tick_right)
         message = BusMessage(moveWheelAtSpeed=MoveWheelAtSpeedMsg(
-            left_tick_per_sec=round(left_speed),
-            right_tick_per_sec=round(right_speed)))
+            left_tick_per_sec=round(tick_left),
+            right_tick_per_sec=round(tick_right)))
         await self._send_message(message)
 
-    async def send_pid(self, pid_left: PIDValues,
-                       pid_right: PIDValues) -> None:
+    async def set_target_positions(self, tick_left: Tick,
+                                   tick_right: Tick) -> None:
         """
-        Sends the PID configurations for both wheels.
+        Sets each wheel's position target, in encoder ticks.
         """
-        LOGGER.get().debug('gateway_send_pid',
-                           pid_left=pid_left,
-                           pid_right=pid_right)
+        LOGGER.get().debug('motor_gateway_set_target_positions',
+                           tick_left=tick_left,
+                           tick_right=tick_right)
+        message = BusMessage(wheelPositionTarget=WheelPositionTargetMsg(
+            tick_left=tick_left, tick_right=tick_right))
+        await self._send_message(message)
 
-        # The motor board uses 16 bit fixed point notation
-        # so the float constants are rounded after a x2^16
-        message = BusMessage(pidConfig=PIDConfigMsg(
-            pid_speed_left=PIDCoefficients(
-                kp=pid_left.k_p * FIXED_POINT_COEF,
-                ki=pid_left.k_i * FIXED_POINT_COEF,
-                kd=pid_left.k_d * FIXED_POINT_COEF,
-            ),
-            pid_speed_right=PIDCoefficients(
-                kp=pid_right.k_p * FIXED_POINT_COEF,
-                ki=pid_right.k_i * FIXED_POINT_COEF,
-                kd=pid_right.k_d * FIXED_POINT_COEF,
-            ),
-        ))
+    # pylint:disable=too-many-arguments
+    async def set_pid_position(self, left_kp: float, left_ki: float,
+                               left_kd: float, right_kp: float,
+                               right_ki: float, right_kd: float) -> None:
+        """
+        Sends the position PID configurations for both wheels.
+        """
+        self.pid_position_left = PIDConstants(left_kp, left_ki, left_kd)
+        self.pid_position_right = PIDConstants(right_kp, right_ki, right_kd)
+
+        LOGGER.get().info('motor_gateway_set_position_pids',
+                          pid_left=self.pid_position_left,
+                          pid_right=self.pid_position_right)
+        await self._send_pid_configs()
+
+    # pylint:disable=too-many-arguments
+    async def set_pid_speed(self, left_kp: float, left_ki: float,
```

Comment:
don't you have a PID entity btw?

Reply:
this is due to limitations on the debug interface made by our lead web dev.

Reply:
hahahaha fair